packages/mongodb/src/MongoDriver.ts

```diff
@@ -138,14 +138,29 @@ export class MongoDriver extends DatabaseDriver<MongoConnection> {
 
   private renameFields<T>(entityName: string, data: T, where = false): T {
     // copy to new variable to prevent changing the T type or doing as unknown casts
-    const copiedData: T & { $fulltext?: string; $text?: { $search: string } } = Object.assign({}, data); // copy first
+    const copiedData: T & { $fulltext?: string; $text?: { $search: string }; $and?: {$fulltext?: string}[] } = Object.assign({}, data); // copy first
```

Comment:
this would introduce merge conficts with v6, lets do it the same [as there](https://github.com/mikro-orm/mikro-orm/blob/v6/packages/mongodb/src/MongoDriver.ts#L170) right ahead

```suggestion
    const copiedData: Dictionary = Object.assign({}, data); // copy first
```